RecoTauTag/HLTProducers/src/HLTTauIPFilter.cc

```diff
@@ -0,0 +1,92 @@
+/** \class HLTTauIPFilter
+ *
+ *  This class is an HLTFilter able to implement
+ *  generic cuts on the tau IP variables available in the PFTauTransverseImpactParameter collection
+ *
+ */
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "DataFormats/TauReco/interface/PFTau.h"
+#include "DataFormats/TauReco/interface/PFTauFwd.h"
+#include "DataFormats/TauReco/interface/PFTauTransverseImpactParameterAssociation.h"
+#include "HLTrigger/HLTcore/interface/HLTFilter.h"
+#include "CommonTools/Utils/interface/StringCutObjectSelector.h"
+#include <vector>
+
+class HLTTauIPFilter : public HLTFilter {
+public:
+  explicit HLTTauIPFilter(const edm::ParameterSet& config);
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  bool hltFilter(edm::Event& event,
+                 const edm::EventSetup& setup,
+                 trigger::TriggerFilterObjectWithRefs& filterproduct) const override;
+
+private:
+  const edm::InputTag m_TausInputTag;
+  const edm::EDGetTokenT<reco::PFTauCollection> m_TausToken;
+  const edm::EDGetTokenT<edm::AssociationVector<reco::PFTauRefProd, std::vector<reco::PFTauTransverseImpactParameterRef>>>
+      m_TausTIPToken;
+
+  const int m_MinTaus;  // min. number of taus required to pass the cuts
+  std::string m_tauTIPSelectorString;
+  StringCutObjectSelector<reco::PFTauTransverseImpactParameter> m_tauTIPSelector;
```

Comment:
```suggestion
  const std::string m_tauTIPSelectorString;
  const StringCutObjectSelector<reco::PFTauTransverseImpactParameter> m_tauTIPSelector;
```
From previous round of comments.